mne/viz/epochs.py

```diff
@@ -65,74 +78,290 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+        else:
+            raise ValueError("If `groupby` is not None, it must be a dict "
+                             "or 'type', got " + type(groupby))
+    return all_picks, all_ch_types, names  # all_picks is a list of lists
+
+
+def _pick_and_combine(epochs, combine, all_picks, all_ch_types, scalings,
+                      names):
+    """Pick and combine epochs image. Helper for plot_epochs_image."""
+    to_plot_list = list()
+    tmin = epochs.times[0]
+
+    if combine is None:
+        epochs = epochs.copy().load_data()
+        for pick, ch_type, _ in zip(all_picks, all_ch_types, names):
+            name = epochs.ch_names[pick]
+            these_epochs = epochs.copy().pick_channels([name])
+            to_plot_list.append([these_epochs, ch_type, name, name])
+    else:
+        from .. import EpochsArray, pick_info
+        data = epochs.get_data()
+        type2name = {"eeg": "EEG", "grad": "Gradiometers",
+                     "mag": "Magnetometers"}
+        combine_title = (" (" + combine + ")"
+                         if isinstance(combine, string_types) else "")
+        if combine == "gfp":
+            combine = lambda data: np.sqrt((data * data).mean(axis=1))  # noqa
+        elif combine == "mean":
+            combine = lambda data: np.mean(data, axis=1)  # noqa
+        elif combine == "std":
+            combine = lambda data: np.std(data, axis=1)  # noqa
+        elif combine == "median":
+            combine = lambda data: np.median(data, axis=1)  # noqa
+        elif not callable(combine):
+            raise ValueError(
+                "`combine` must be None, a callable or one out of 'mean' "
+                "or 'gfp'. Got " + str(type(combine)))
+        for ch_type, picks_, name in zip(all_ch_types, all_picks, names):
+            if len(np.atleast_1d(picks_)) < 2:
+                raise ValueError("Cannot combine over only one sensor. "
+                                 "Consider using different values for "
+                                 "`picks` and/or `groupby`.")
+            if ch_type == "grad":
+                def pair_and_combine(data):
+                    data = data ** 2
+                    data = (data[:, ::2, :] + data[:, 1::2, :]) / 2
+                    return combine(np.sqrt(data))
+                picks_ = _grad_pair_pick_and_name(epochs.info, picks_)[0]
+                this_data = pair_and_combine(
+                    data[:, picks_, :])[:, np.newaxis, :]
+            else:
+                this_data = combine(
+                    data[:, picks_, :])[:, np.newaxis, :]
+            info = pick_info(epochs.info, [picks_[0]], copy=True)
+            these_epochs = EpochsArray(this_data.copy(), info, tmin=tmin)
+            d = these_epochs.get_data()  # Why is this necessary?
+            d[:] = this_data  # Without this, the data is all-zeros!
+            to_plot_list.append([these_epochs, ch_type, name,
+                                 type2name.get(name, name) + combine_title])
```

Comment:
Or maybe unindent this to unify with the case of ``combine is None``. There are options, but looks like this function can be shortened a bit.